mbf_abstract_nav/include/mbf_abstract_nav/abstract_action_base.hpp

```diff
@@ -175,11 +179,12 @@ class AbstractActionBase
   {
     uint8_t slot = goal_handle.getGoal()->concurrency_slot;
 
-    boost::lock_guard<boost::mutex> guard(slot_map_mtx_);
+    slot_map_mtx_.lock();
     typename ConcurrencyMap::iterator slot_it = concurrency_slots_.find(slot);
```

Comment:
just as an idea: would it make sense to write a function which looks like
```
typename ConcurrencyMap::iterator AbstractActionBase::findSlot(uint_8 _key) {
 boost::scoped_lock<...> lock;
return concurrency_slots.find(_key)
}

```

Reply:
yes, I like it